### PR TITLE
updates to resource path

### LINKED ISSUE #4313: The `resource-path` configuration applies to many (all?) resource types
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/resource-configs/resource-path

### What part(s) of the page would you like to see updated?

The [`resource-path`](https://docs.getdbt.com/reference/resource-configs/resource-path) page only explains how it applies to:
- models
- seeds
- snapshots

But it applies to other [resource types](https://docs.getdbt.com/docs/dbt-cloud-apis/project-state#affected-states-by-node-type) as well:
- tests
- sources
- potentially others

### The problem

https://github.com/dbt-labs/dbt-core/issues/8876 specifically mentioned that they were expecting this section to apply to `tests` also:

> The `<resource-path>` nomenclature is used in this documentation when documenting how to configure a model, seed, or snapshot, from your `dbt_project.yml` file. It represents the nested dictionary keys that provide the path to either a directory of models, or a single model.

The "info" section is also relevant:

> This example is for models, but the same concepts apply for seeds and snapshots.

But these sections only explain three node types and don't cover the other ones.

### Potential solution

The key insight is that it should be something like this instead:

> It represents the nested dictionary keys that provide the path to a directory of that resource type, or a single instance of that resource type by name.

e.g.

```yaml
<resource type>:
  <project name>:
    <directory name>:
      <subdirectory name>:
        <instance of resource type (by name)>:
          ...
```

### Related issues
- https://github.com/dbt-labs/dbt-core/issues/8876
- https://github.com/dbt-labs/dbt-core/issues/8960

### PR DESCRIPTION
**Pull request by mirnawong1**
PR clarifies how to disable sources and addresses core issue https://github.com/dbt-labs/dbt-core/issues/8960 and adds the resource paths page to the sidebar.

This also Resolves #4313

### CODE REVIEW

**Comment by mirnawong1** [comment]
Waiting on @graciegoheen  and will follow up this week

**Review by graciegoheen — APPROVED** [review]
THANK YOU!

File: website/docs/reference/resource-configs/resource-path.md [hunk 1]
```diff
@@ -1,11 +1,28 @@
-The `<resource-path>` nomenclature is used in this documentation when documenting how to configure a model, seed, or snapshot, from your `dbt_project.yml` file. It represents the nested dictionary keys that provide the path to either a directory of models, or a single model.
+---
+title: Resource path
+description: "Learn how to use resource paths to configure resource types in dbt."
+id: resource-path
+sidebar_label: "About resource paths"
+---
+
+The `<resource-path>` nomenclature is used in this documentation when documenting how to configure resource types like models, seeds, snapshots, tests, sources, and others, from your `dbt_project.yml` file.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/resource-path.md:8]:
```suggestion
The `resource-path` nomenclature is used in this documentation when documenting how to configure resource types like models, seeds, snapshots, tests, sources, and others, from your `dbt_project.yml` file. 
```

File: website/docs/reference/resource-configs/resource-path.md [hunk 2]
```diff
@@ -1,11 +1,28 @@
-The `<resource-path>` nomenclature is used in this documentation when documenting how to configure a model, seed, or snapshot, from your `dbt_project.yml` file. It represents the nested dictionary keys that provide the path to either a directory of models, or a single model.
+---
+title: Resource path
+description: "Learn how to use resource paths to configure resource types in dbt."
+id: resource-path
+sidebar_label: "About resource paths"
+---
+
+The `resource-path` nomenclature is used in this documentation when documenting how to configure resource types like models, seeds, snapshots, tests, sources, and others, from your `dbt_project.yml` file.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/resource-path.md]:
```suggestion
The `<resource-path>` nomenclature is used in this documentation when documenting how to configure resource types like models, seeds, snapshots, tests, sources, and others, from your `dbt_project.yml` file. 
```

**Pull request merged**